File: src/services/Job/Job.Domain/Entities/Job.cs
using Job.Domain.Enums;

namespace Job.Domain.Entities;

public class Job
{
    public Guid Id { get; set; }
    
    public JobType Type { get; set; }
    public JobStatus Status { get; set; } = JobStatus.Bekliyor;
    
    public string? SourceDocument { get; set; }
    
    public Guid? AssignedOperatorId { get; set; }
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? FinishedAt { get; set; }

    public List<JobItem> Items { get; set; } = new();

}

File: frontend/types/job.ts
export interface JobItem {
    id: string;
    productId: string;
    productName: string;
    sourceLocationId?: string;
    targetLocationId?: string;
    quantity: number;
}

export interface Job {
    id: string;
    type: string;
    status: string;
    sourceDocument: string;
    createdAt: string;
    items: JobItem[];
}

export enum JobType {
    Yerleştirme = 0,
    Toplama = 1,
    Transfer = 2,
    Sayım = 3
}

export interface CreateJobItemRequest {
    productId: string;
    quantity: number;
    sourceLocationId?: string;
    targetLocationId?: string;
}

export interface CreateJobRequest {
    type: JobType;
    sourceDocument: string;
    items: CreateJobItemRequest[];
}